specs/pluggable_alert_delivery.adoc

```diff
@@ -0,0 +1,370 @@
+                    notification is dispatched to act as source(from) of all
+                    mail notifications.
+              ***** Presence of all mandatory fields in config mandated by the
+                    respective handler.
+
+----
+  Note:
+    * The different alerts need to be yet classified on various importance
+      levels.
+      And this is my current opinion:
+      **  Any threshold breach corresponding notification is a 'HIGH'
+          significance alerts
+    * The complete list of alerts and their importance/significance level needs
+      to be decided.Following are some of the github issues to capture the list
+      ** https://github.com/Tendrl/documentation/issues/44
+      ** https://github.com/Tendrl/documentation/issues/45
+      ** https://github.com/Tendrl/documentation/issues/46
+----
+
+
+=== Alternatives
+
+To be explored.
+
+=== Data model impact
+
+The structure of alert can be the following:
+
+----
+{
+  'alert-id': <unique tendrl generated id>,
+  'node-id': <id of node on which alert was detected>,
+  'time-stamp': <time stamp of alert>,
+  'resource': <the name of resource for which alert has been raised>,
+  'current-value': <the current observed value status/utilization as applies>
+  'tags': <custom alert specific info>,
+  'type': <the type of alert percent-used/status of resource>,
+  'severity': <severity of alert>,
+  'significance': <the severity of importance of notifying the alert>,
+  'ackedby': <indicate who acked the alert>,
+  'acked': <boolean to indicate if the alert is acked>,
+}
+----
+
+ex:
+For a performance monitoring related alert:
+
+----
+{
+  alert_id : '6405962e-bc46-11e6-a4a6-cec0c932ce01',
+  node_id: '5205962e-bc46-11e6-a4a6-cec0c932cz01',
+  time_stamp: '1481046935.536',
+  'resource': 'memory',
+  'CurrentValue': '7.176942e+00',
+  tags: {
+    'WarningMax': '1.000000e+00',
+    'FailureMax': '2.000000e+00',
+  },
+  'Type': 'percent-used',
+  'severity': 'Critical',
+  'ackedby': '',
+  'acked': False,
+  'significance': 'HIGH',
+}
+----
+
+and for a status based alert:
+
+----
+{
+  alert_id : '6405962e-bc46-11e6-a4a6-cec0c932ce01',
+  node_id: '5205962e-bc46-11e6-a4a6-cec0c932cz01',
+  time_stamp: '1481046935.536',
+  'resource': 'cluster',
+  'CurrentValue': 'Down',
+  tags: {
+    'cluster_id' : '6406062e-be46-11e6-a4a6-cec0c932ce01',
+  },
+  'Type': 'status',
+  'severity': 'Critical',
+  'ackedby': '',
+  'acked': False,
+  'significance': 'HIGH',
+}
+----
+
+The severity levels can be 'Critical', 'Info' or 'Warning'.
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The tendrl api needs to proxy to apis exposed by Tendrl/alerting as mentioned
+in section below(Notifications/Monitoring impact)
+
+==== Notifications/Monitoring impact:
+
+Tendrl/alerting needs to implement flows for apis as described in section
+"Tendrl API impact".
+
+The flow definition for the above will look like:
```

Comment:
For cluster context alerts, the cluster context will be added to the tags attribute which is currently wrongly mentioned as of type string. It is instead supposed to be json. And the framework is otherwise agnostic of whether its a cluster specific alert or just a node-apecific alert